src/Symfony/Component/Workflow/Configuration/AttributeReader.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Configuration;
+
+use Symfony\Component\Workflow\Attribute\AsWorkflow;
+use Symfony\Component\Workflow\Attribute\Transition;
+use Symfony\Component\Workflow\WorkflowType;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class AttributeReader
+{
+    public function extractConfiguration(AsWorkflow $attribute, \ReflectionClass $reflection): array
+    {
+        return [
+            'name' => $attribute->name,
+            'type' => WorkflowType::StateMachine === $attribute->type ? 'state_machine' : 'workflow',
+            'marking_store' => $attribute->markingStore ?: ['type' => 'method', 'property' => 'marking'],
+            'supports' => $attribute->supports,
+            'places' => $attribute->places,
+            'transitions' => $this->extractTransitions($reflection),
+            'metadata' => [],
+            'audit_trail' => [
+                'enabled' => $attribute->auditTrail,
+            ],
+            'initial_marking' => [],
+            'events_to_dispatch' => null,
+        ];
+    }
+
+    private function extractTransitions(\ReflectionClass $reflection): array
+    {
+        $transitions = [];
+
+        foreach ($reflection->getReflectionConstants() as $constant) {
+            $transitionAttributes = $constant->getAttributes(Transition::class);
+
+            if (!$transitionAttributes) {
+                continue;
+            }
+
+            $transition = $transitionAttributes[0]->newInstance();
```

Comment:
this should read **all** transition attributes, not only the first one. Otherwise, this won't allow defining multiple transitions with the same name but distinct `froms` in a Workflow (which is valid in the component, even though I'm not sure it is fully supported in all the dumpers)